soccer/src/soccer/planning/planner_node.hpp

```diff
@@ -162,6 +163,18 @@ class PlannerForRobot {
      */
     void execute_trajectory(const RobotIntent& intent);
 
+    /*
+     * @brief estimate the amount of time it would take for a robot to execute a robot intent
+     * SERVICE).
+     *
+     * @param request Requested RobotIntent resulting in the hypothetical robot path.
+     * @param response The response object that will contain the resultant time to completion of a
+     * hypothetical path.
+     */
+    void plan_hypothetical_robot_path(
+        const std::shared_ptr<rj_msgs::srv::PlanHypotheticalPath::Request>& request,
+        std::shared_ptr<rj_msgs::srv::PlanHypotheticalPath::Response>& response);
```

Comment:
This is up for debate, but if you want to get around this check warning, simply pass the raw ptr stored inside the shared_ptr here. So something like 
```
plan_hypothetical_robot_path(request, response.get());
```

and 
```
void PlannerForRobot::plan_hypothetical_robot_path(
    const std::shared_ptr<rj_msgs::srv::PlanHypotheticalPath::Request>& request,
    rj_msgs::srv::PlanHypotheticalPath::Response* raw_response) {
```
LearnCpp advocates for this when dealing with unique ptrs, to avoid passing ownership when copying by value (but for shared ptrs this is not a big issue): https://www.learncpp.com/cpp-tutorial/stdunique_ptr/

This StackOverflow post seems to have a few varying opinions on the topic of passing shared_ptr to funcs: https://stackoverflow.com/questions/10826541/passing-shared-pointers-as-arguments

Reply:
Actually, I have decided that I dislike this check warning. It is based on an outdated version of the Google style guide: https://releases.llvm.org/3.9.0/tools/clang/tools/extra/docs/clang-tidy/checks/google-runtime-references.html

The new Google style guide says input/output params are fine: https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs

Reply:
patched in PR #1965